cluster/local/provider.Tiltfile

```diff
@@ -0,0 +1,196 @@
+#
+# 1) core crossplane repo
+# 2) any standalone provider
+####################################################################################
+
+# Register CRDs which expose `image` for Tilt to be able to find them and
+# build the corresponding docker image for it.
+# https://docs.tilt.dev/custom_resource.html#does-it-contain-images
+k8s_kind('ControllerConfig', image_json_path = '{.spec.image}')
+
+# Build a crossplane provider
+#
+# 1. build and apply CRDs
+# 2. build and deploy controller manager
+def build_provider(name, provider, settings):
+    _generate_crds(provider, settings)
+    _build_controller(name, provider, settings)
+
+# Generate and apply crossplane provider's CRDs
+def _generate_crds(provider, settings):
+    context = provider.get('context')
+
+    # Prefix each dependency with context so Tilt can watch the correct paths
+    # for changes. At the same time keep track of 'zz_generated.*' files to
+    # ignore (this is needed to prevent being stuck in an infinite loop.)
+    crd_deps = []
+    ignore_deps = []
+    for d in provider.get('crd_deps', []):
+        crd_deps.append(context + '/' + d)
+        ignore_deps.append(context + '/' + d + '/**/zz_generated.*')
+        
+    # Generate crossplane provider CRDs out of API definition
+    local_resource(
+        'gen-crds-' + provider['short_name'],
+        'cd ' + context + ' && make generate',
+        deps = crd_deps,
+        ignore = ignore_deps
+    )
+
+    # Find all the CRDs manifests files
+    crds_folder = context + '/' + provider.get('crds_folder', 'package/crds')
+    files = listdir(crds_folder)
+
+    # Extract name of the CRDs to show on Tilt UI
+    crds_name = []
+    for f in files:
+        crds_name.append(read_yaml(f)['metadata']['name'] + ':customresourcedefinition')
+
+    # Apply CRD manifests
+    k8s_yaml(files)
+    k8s_resource(
+        new_name = 'crds-' + provider['short_name'],
+        objects = crds_name,
+        resource_deps = settings.get('resource_deps')
+    )
+
+# Build crossplane provider binary and docker image. Binary is built for
+# linux-amd64 (to be only used inside a container.) The binary is stored
+# in crossplane cloned folder at _output/tilt/
+def _build_controller(name, provider, settings):
+    context = provider.get('context')
+    go_main = provider.get('go_main')
+    core_path = settings.get('core_path')
+
+    # Prefix each live reload dependency with context so Tilt
+    # can watch the correct paths for changes
+    cmd_deps = []
+    ignore_deps = []
+    for d in provider.get('cmd_deps', []):
+        cmd_deps.append(context + '/' + d)
+        ignore_deps.append(context + '/' + d + '/**/zz_generated.*')
+
+    # Build crossplane provider
+    build_output = '_output/tilt/' + name
+    local_resource(
+        'build-' + provider['short_name'],
+        'cd ' + context + ' && CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o ' + core_path + '/' + build_output + ' ' + go_main,
```

Comment:
Is there an advantage here to just using `make build` and copying over to the necessary directory if necessary? That would ensure consistency between development and build.

Reply:
That would make an assumption that `make build` exists and it builds with `CGO_ENABLED=0 GOOS=linux GOARCH=amd64` (to be run inside an `alpine` Docker image). Assuming if `Makefile` and `build` target exist is not very far fetched (to be honest), but not so much for the `GOOS` and `GOARCH`.

Also `make build` (as it stands now) runs some slightly irrelevant checks (for Tilt context) such as module dependencies verification, golangci-lint, to name a few. I mean they are completely important but not necessarily in Tilt when you want to live reload as fast as possible (under a second or so) when you change a Go file.